sdk/cosmos/azure-cosmos-tests/src/test/java/com/azure/cosmos/EndToEndTimeOutWithAvailabilityTest.java

```diff
@@ -85,17 +87,31 @@ public void readItemWithEndToEndTimeoutPolicyInOptionWithSpeculationShouldNotTim
 
         CosmosItemRequestOptions options = new CosmosItemRequestOptions();
         options.setCosmosEndToEndOperationLatencyPolicyConfig(endToEndOperationLatencyPolicyConfig);
-        TestObject itemToRead = createdDocuments.get(random.nextInt(createdDocuments.size()));
+        EndToEndTimeOutValidationTests.TestObject itemToRead = createdDocuments.get(random.nextInt(createdDocuments.size()));
         FaultInjectionRule rule = injectFailure(createdContainer, FaultInjectionOperationType.READ_ITEM, null);
 
-        Mono<CosmosItemResponse<TestObject>> cosmosItemResponseMono =
-            createdContainer.readItem(itemToRead.id, new PartitionKey(itemToRead.mypk), options, TestObject.class);
+        Mono<CosmosItemResponse<EndToEndTimeOutValidationTests.TestObject>> cosmosItemResponseMono =
+            createdContainer.readItem(itemToRead.getId(), new PartitionKey(itemToRead.getMypk()), options, EndToEndTimeOutValidationTests.TestObject.class);
 
+        verifyExpectError(cosmosItemResponseMono);
+
+        // Now try the same request with Threshold based availability strategy
+        CosmosEndToEndOperationLatencyPolicyConfig config = new CosmosEndToEndOperationLatencyPolicyConfigBuilder(Duration.ofSeconds(3))
+            .setAvailabilityStrategy(new ThresholdBasedAvailabilityStrategy(Duration.ofMillis(100), Duration.ofMillis(200), 2))
+            .build();
+        cosmosItemResponseMono =
```

Comment:
the new config is not being set on the option, so the request should still fail?

Reply:
correct options still has the default policy form the factory - should probably be adjusted

Reply:
fixed